app/utils/nonsipp/CheckStatusUtils.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.nonsipp
+
+import models.IdentityType._
+import models.SchemeHoldLandProperty._
+import config.RefinedTypes.{Max5000, OneTo5000}
+import models.SchemeId.Srn
+import pages.nonsipp.landorproperty._
+import eu.timepit.refined.refineV
+import models.{IdentitySubject, IdentityType, UserAnswers}
+import pages.nonsipp.common._
+import models.IdentitySubject._
+
+object CheckStatusUtils {
+
+  /**
+   * This method determines whether or not the Land or Property section needs to be checked. A section needs to be
+   * checked if 1 or more records in that section need to be checked.
+   * @param userAnswers the answers provided by the user, from which we get each Land or Property record
+   * @param srn the Scheme Reference Number, used for the .get calls
+   * @return true if any record requires checking, else false
+   */
+  def checkLandOrPropertySection(
+    userAnswers: UserAnswers,
+    srn: Srn
+  ): Boolean = {
+    val landOrPropertyHeld = userAnswers.get(LandOrPropertyHeldPage(srn))
+    val journeysStartedList = userAnswers.get(LandPropertyInUKPages(srn)).getOrElse(Map.empty).keys.toList
+
+    landOrPropertyHeld match {
+      case None => false
+      case Some(false) => false
+      case Some(true) =>
+        journeysStartedList
+          .map(
+            index => {
+              refineV[OneTo5000](index.toInt + 1).fold(
+                _ => List.empty,
+                refinedIndex => checkLandOrPropertyRecord(userAnswers, srn, refinedIndex)
+              )
+            }
+          )
+          .contains(true)
+    }
+  }
+
+  /**
+   * This method determines whether or not a Land or Property record needs to be checked. A record needs checking if
+   * all of the pre-populated-then-cleared answers are missing & all of the other answers are present.
```

Comment:
wouldn't the logic need to be
```
A record needs checking if ANY of the pre-populated-then-cleared answers are missing
```

Reply:
For LOP, I think after the pre-populated-then-cleared `userAnswers.get(LandOrPropertyHeldPage(srn))` will be empty and that means we need to check here, don't we ?

https://docs.google.com/spreadsheets/d/1ZeT7B6xhryMHjE231WCW_N6MXypo04x0QyzFEKAoraU/edit?pli=1&gid=0#gid=0

Reply:
I think the rationale behind the 'all' implementation is that we're less likely to show any newly-added-but-incomplete journeys as records that need checking, since the user would have to back out of the journey at exactly the right page for that to happen. If we use 'any' rather than 'all', it's more likely that a newly-added-but-incomplete journey will end up being displayed as needing checking.

The drawback of this is that if a user goes to check a pre-populated journey and only provides partial answers, then backs out, then this journey will no longer be displayed as needing checking. This was deemed unlikely to happen, so the drawbacks of the 'all' implementation were deemed to be less likely to occur than the drawbacks of the 'any' implementation.

Neither implementation is ideal, which is why I thought it would be worth looking into another suggested solution, but that wasn't possible within the given timeframe, so for now the decision was made to take the 'all' approach.

Reply:
@cyeniceri That will be checked by the `checkLandOrPropertySection` method when the user lands on the Task List page, whereas this method will be called for each method on the new LOP List page implementation, at which point we already know that the user has some LOP and so `LandOrPropertyHeldPage` must be `true`.